x/proof/keeper/msg_server_submit_proof.go

```diff
@@ -1,17 +1,35 @@
 package keeper
 
 import (
+	"bytes"
 	"context"
+	"crypto/sha256"
 
+	"github.com/pokt-network/smt"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 
+	"github.com/pokt-network/poktroll/pkg/relayer/protocol"
 	"github.com/pokt-network/poktroll/x/proof/types"
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+	sessionkeeper "github.com/pokt-network/poktroll/x/session/keeper"
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+)
+
+const (
+	// TODO_TECHDEBT: relayMinDifficultyBits should be a governance-based parameter
+	relayMinDifficultyBits = 0
+
+	// sumSize is the size of the sum of the relay request and response
```

Comment:
`is the size of the sum of the relay request and response`

What do you mean by this?

I don't think it has anything to do with the relay request and response and is actually a function of the SMT leaf. I may be misunderstanding.

Reply:
We currently have no SMT function that extracts the relay from the Proof's `ClosestValueHash` which contains both the sum and the relay bytes. We need this value to be able to truncate the `closestValueHash`. 

It is currently mimicking what's happening here https://github.com/pokt-network/smt/blob/main/proofs.go#L319.

While the `sumSize` is defined here https://github.com/pokt-network/smt/blob/main/types.go#L10.

We can get rid of this when the SMT package offer a method to extract the relay as stated in the `TODO_TECHDEBT` below.